src/IntegrationTests/Applications/SampleAspNetCoreApp/Startup.cs

```diff
@@ -47,6 +47,8 @@ public void ConfigureServices(IServiceCollection services)
                 var dependenciesCollector = new DependenciesCollector(new HttpClientCollectorOptions(), tracerFactory);
                 var aspNetCoreCollector = new AspNetCoreCollector(tracerFactory.GetTracer(null));
             });
+
```

Comment:
If you were going to dependency inject the data sender, you would declare it here.
then the CountMetricGenerator would take the datasender in its constructor and the dependency injection container is responsible for resolving it.